packages/ecs-agent/ecs-agent.spec

```diff
@@ -0,0 +1,89 @@
+%global goproject github.com/aws
+%global gorepo amazon-ecs-agent
+%global goimport %{goproject}/%{gorepo}
+
+%global gover 1.41.0
+
+Name: %{_cross_os}ecs-agent
+Version: %{gover}
+Release: 1%{?dist}
+Summary: Amazon Elastic Container Service agent
+License: Apache-2.0
+URL: https://%{goimport}
+Source0: https://%{goimport}/archive/v%{gover}.tar.gz
+Source1: ecs.service
+Source2: pause-config.json
+Source3: pause-manifest.json
+Source4: pause-repositories
+
+# Upstream: https://github.com/aws/amazon-ecs-agent/pull/2513
+Patch0001: 0001-engine-move-default-image-exclusions.patch
+
+# Bottlerocket-specific - filesystem location of the pause image
+Patch0002: 0002-bottlerocket-default-filesystem-locations.patch
+
+BuildRequires: %{_cross_os}glibc-devel
+
+Requires: %{_cross_os}docker-engine
+# for sysctl
+Requires: %{_cross_os}procps
+Requires: %{_cross_os}iptables
+
+%description
+%{summary}.
+
+%prep
+%autosetup -Sgit -n %{gorepo}-%{gover} -p1
+%cross_go_setup %{gorepo}-%{gover} %{goproject} %{goimport}
+
+%build
+# Build the agent
+%cross_go_configure %{goimport}
+PAUSE_CONTAINER_IMAGE_NAME="amazon/amazon-ecs-pause"
+PAUSE_CONTAINER_IMAGE_TAG="bottlerocket"
+LD_PAUSE_CONTAINER_NAME="-X github.com/aws/amazon-ecs-agent/agent/config.DefaultPauseContainerImageName=${PAUSE_CONTAINER_IMAGE_NAME}"
+LD_PAUSE_CONTAINER_TAG="-X github.com/aws/amazon-ecs-agent/agent/config.DefaultPauseContainerTag=${PAUSE_CONTAINER_IMAGE_TAG}"
+go build -a \
+  -buildmode pie \
+  -ldflags "${LD_PAUSE_CONTAINER_NAME} ${LD_PAUSE_CONTAINER_TAG}" \
+  -o amazon-ecs-agent \
+  ./agent
+
+# Build the pause container
+(
+  set -x
+  cd misc/pause-container/buildPause
+  mkdir -p rootfs/usr/bin
+  # TODO determine which other C and LD flags we want
+  make BIN=rootfs/usr/bin/pause GCC=%{_cross_triple}-musl-gcc
+
+  # Construct image
+  mkdir -p image/rootfs
+  tar cvf image/rootfs/layer.tar -C rootfs .
+  DIGEST=$(sha256sum image/rootfs/layer.tar | sed -e 's/ .*//')
+  echo -n "1.0" > image/rootfs/VERSION
+  install -m 0644 %{S:2} image/config.json
+  sed -i "s/~~digest~~/${DIGEST}/" image/config.json
+  install -m 0644 %{S:3} image/manifest.json
+  install -m 0644 %{S:4} image/repositories
+  tar cvf ../../../amazon-ecs-pause.tar -C image .
+)
```

Comment:
If possible it might be nice to build this in a reproducible way. I don't know if GCC's output will be reproducible, but making the timestamps + order of files in the tarball reproducible is definitely doable: https://reproducible-builds.org/docs/archives/

Reply:
Done